lib/libutee/include/pta_attestation.h

```diff
@@ -0,0 +1,99 @@
+ *                           attacks)
+ * [out]    memref[2]        Output buffer. Receives the signed digest.
+ *                           - The first 32 bytes are the digest itself (from
+ *                             the TA signed header: struct shdr::hash)
+ *                           - The following bytes are a signature:
+ *                               SIG(SHA256(Nonce | digest))
+ *                           - The algorithm is
+ *                             TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA256 with a salt
+ *                             length of 32.
+ *                           - The key pair is generated internally and stored
+ *                             in secure storage. The public key can be
+ *                             retrieved with command PTA_ATTESTATION_GET_PUBKEY
+ *                             (typically during device provisioning).
+ *                           Given that the sigature length is equal to the
+ *                           RSA modulus size in bytes, the output buffer size
+ *                           should be at least (digest size + modulus size)
+ *                           bytes. For example, for a 32-byte SHA256 digest and
+ *                           2048 bit key (256 bytes) the minimum buffer size is
+ *                           288 bytes.
+ *
+ * Return codes:
+ * TEE_SUCCESS
+ * TEE_ERROR_BAD_PARAMETERS - Incorrect input param
+ * TEE_ERROR_SHORT_BUFFER - Output buffer size less than required
+ */
+#define PTA_ATTESTATION_GET_TA_SHDR_DIGEST 0x1
+
+/*
+ * Return a signed hash for a running user space TA, which must be the caller
+ * of this PTA. It is a runtime measurement of the memory pages that contain
+ * immutable data (code and read-only data).
+ *
+ * [in]     memref[0]        Nonce
+ * [out]    memref[1]        SHA256 hash of the TA memory followed by a
+ *                           signature. See PTA_ATTESTATION_GET_TA_HDR_DIGEST
+ *                           for a description of the signature.
+ *
+ * Return codes:
+ * TEE_SUCCESS
+ * TEE_ERROR_ACCESS_DENIED - Caller is not a user space TA
+ * TEE_ERROR_BAD_PARAMETERS - Incorrect input param
+ * TEE_ERROR_SHORT_BUFFER - Output buffer size less than required
+ */
+#define PTA_ATTESTATION_HASH_TA_MEMORY 0x2
+
+/*
+ * Return a signed hash of the TEE OS (kernel) memory. It is a runtime
+ * measurement of the memory pages that contain immutable data (code and
+ * read-only data).
+ *
+ * [in]     memref[0]        Nonce
+ * [out]    memref[1]        SHA256 hash of the TEE memory followed by a
+ *                           signature. See PTA_ATTESTATION_GET_TA_HDR_DIGEST
+ *                           for a description of the signature.
+ *
+ * Return codes:
+ * TEE_SUCCESS
+ * TEE_ERROR_ACCESS_DENIED - Caller is not a user space TA
+ * TEE_ERROR_BAD_PARAMETERS - Incorrect input param
+ * TEE_ERROR_SHORT_BUFFER - Output buffer size less than required
+ */
+#define PTA_ATTESTATION_HASH_TEE_MEMORY 0x3
```

Comment:
Here as well, we should have an output value parameter to say that its an RSA signature.